tests/unit_tests/output_selection.cpp

```diff
@@ -109,28 +109,48 @@ TEST(select_outputs, order)
     offset = n_outs += (n); \
   }
 
-TEST(select_outputs, gamma)
+void pick_outputs(size_t npicks, std::vector<uint64_t> offsets, size_t n_outs, tools::gamma_picker picker, double &median)
 {
-  std::vector<uint64_t> offsets;
-
-  MKOFFSETS(300000, 1);
-  tools::gamma_picker picker(offsets);
-  std::vector<double> ages(100000);
+  std::vector<double> ages(npicks);
   double age_scale = 120. * (offsets.size() / (double)n_outs);
   for (size_t i = 0; i < ages.size(); )
   {
     uint64_t o = picker.pick();
-    if (o >= n_outs)
+    bool acceptable_bad_pick = o == std::numeric_limits<uint64_t>::max();
```

Comment:
Minor style issue, but I think using variable assignments in place of comments only further confuses things:
1. A variable assignment implies the need for repeated references, but this is a throw-away expression which results in a continue.
2. The exact syntax in this situation (with many equal signs near each other) proves difficult to read, at least for me.

If you find max as a placeholder for "no real pick" confusing, then write a clarifying comment. The variable name alone doesn't work, because it says, in effect, the same thing as the code would without the variable: "Allow this max pick"; what it doesn't say, which may or may not be self-evident, is why such a choice should be allowed.